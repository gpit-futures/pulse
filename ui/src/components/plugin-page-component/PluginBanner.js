import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { themeConfigs } from '../../themes.config';

export default class PluginBanner extends PureComponent {
  static propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string.isRequired,
    img: PropTypes.any.isRequired,
  };

  static defaultProps = {
    toRight: false,
  };

  render() {
    const { title, subTitle, img, toRight, isShowPluginsBigBanner } = this.props;

    return (
      themeConfigs.isShowPluginsBigBanner || isShowPluginsBigBanner ?
        <div className="page-banner">
          <div className={`page-banner__title ${toRight ? 'page-banner__title--right' : ''}`}>{title}</div>
          <div className={`page-banner__subtitle ${toRight ? 'page-banner__subtitle--right' : ''}`}>{subTitle}</div>
          <img className="page-banner__img" src={img} alt="" />
        </div>
        : <div />
    )
  }
}
