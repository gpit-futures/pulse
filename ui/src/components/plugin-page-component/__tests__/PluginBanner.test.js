import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';

import PluginBanner from '../PluginBanner';

Enzyme.configure({ adapter: new Adapter() });

describe('Component <PluginBanner />', () => {
  it('should renders with all props correctly', () => {
    const component = shallow(
      <PluginBanner
        isShowPluginsBigBanner
        toRight={false}
        title="test title"
        subTitle="test subTitle"
        img="test img"
      />);

    expect(component.find('.page-banner')).toHaveLength(1);
    expect(component.find('img')).toHaveLength(1);
    expect(component.find('.page-banner__title').text()).toEqual('test title');
    expect(component.find('.page-banner__subtitle').text()).toEqual('test subTitle');

    expect(component).toMatchSnapshot();

    component.setProps({ toRight: true });
    expect(component.find('.page-banner__title--right')).toHaveLength(1);
    expect(component.find('.page-banner__subtitle--right')).toHaveLength(1);

    expect(component).toMatchSnapshot();

    component.setProps({ isShowPluginsBigBanner: false });
    expect(component.find('.page-banner')).toHaveLength(0);

    expect(component).toMatchSnapshot();
  });
});
