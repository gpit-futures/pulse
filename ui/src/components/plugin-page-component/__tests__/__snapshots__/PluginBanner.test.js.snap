// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <PluginBanner /> should renders with all props correctly 1`] = `
<div
  className="page-banner"
>
  <div
    className="page-banner__title "
  >
    test title
  </div>
  <div
    className="page-banner__subtitle "
  >
    test subTitle
  </div>
  <img
    alt=""
    className="page-banner__img"
    src="test img"
  />
</div>
`;

exports[`Component <PluginBanner /> should renders with all props correctly 2`] = `
<div
  className="page-banner"
>
  <div
    className="page-banner__title page-banner__title--right"
  >
    test title
  </div>
  <div
    className="page-banner__subtitle page-banner__subtitle--right"
  >
    test subTitle
  </div>
  <img
    alt=""
    className="page-banner__img"
    src="test img"
  />
</div>
`;

exports[`Component <PluginBanner /> should renders with all props correctly 3`] = `<div />`;
